packages/react/src/hooks/useAuth.ts

```diff
@@ -27,32 +23,64 @@ export const useAuth = (): UseAuthResult => {
     user: undefined,
   });
 
-  const handleAuth: HubCallback = ({ payload }) => {
-    switch (payload.event) {
-      case 'signIn':
-        return setResult({ user: payload.data as AuthUser, isLoading: false });
-      case 'signOut':
-        return setResult({ isLoading: false });
-      default:
-        break;
+  /**
+   * Hub events like `tokenRefresh` will not give back the user object.
+   * This util will be used to get current user after those events.
+   */
+  const fetchCurrentUser = React.useCallback(async () => {
+    setResult({ user: undefined, isLoading: true, error: undefined });
+
+    try {
+      const user = (await Auth.currentAuthenticatedUser()) as AmplifyUser;
```

Comment:
`Auth.currentAuthenticatedUser()` returns `any`, so I need to cast it.

Reply:
Maybe we can put the above comment in the code 😆

Reply:
😈 
```suggestion
     // casting the result because `Auth.currentAuthenticateduser` returns `any`
      const user = (await Auth.currentAuthenticatedUser()) as AmplifyUser;
```